### PR TITLE
[angular-xmcloud] Introduce column-splitter component for angular

### PR DESCRIPTION
**Pull request by sc-addypathania**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated
- [ ] Upgrade guide entry added

## Description / Motivation
Add `ColumnSplitterComponent` to angular xmcloud

## Testing Details

- [ ] Unit Test Added
- [x] Manual Test/Other (Please elaborate)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    super.ngOnInit();
+  }
+
+  get columnSplitterStyles(): string {
+    return `${this.rendering.params.GridParameters ?? ''} ${this.rendering.params.Styles ??
+      ''}`.trimEnd();
+  }
+
+  get columnWidths(): string[] {
+    return [
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.ts:19]:
```suggestion
    return Array.from({ length: 8 }, (_, i) => this.rendering.params[`ColumnWidth${i + 1}`]);
```

File: packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      this.rendering.params.ColumnWidth4,
+      this.rendering.params.ColumnWidth5,
+      this.rendering.params.ColumnWidth6,
+      this.rendering.params.ColumnWidth7,
+      this.rendering.params.ColumnWidth8,
+    ];
+  }
+
+  get columnStyles(): string[] {
+    return [
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.ts:32]:
```suggestion
    return Array.from({ length: 8 }, (_, i) => this.rendering.params[`Styles${i + 1}`]);
```

File: packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+﻿import { Component, OnInit } from '@angular/core';
+import { SxaComponent } from '../sxa.component';
+
+@Component({
+  selector: 'app-column-splitter',
+  templateUrl: './column-splitter.component.html',
+})
+export class ColumnSplitterComponent extends SxaComponent implements OnInit {
+  ngOnInit() {
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.ts:9]: ngOnInit is not needed if you are not defining something there, the same for "OnInit" interface

File: packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿<div [ngClass]="['row', 'component', 'column-splitter', columnSplitterStyles]" [attr.id]="id || null">
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.html:1]: "class" attribute is not computed, so it doesn't make sense to use "[ngClass]"
Do we need to set a fallback value for "id" to "null"? Do we even need to set a fallback value?
```suggestion
﻿<div class="row component column-splitter columnSplitterStyles" [attr.id]="id || null">
```

File: packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  <div *ngFor="let ph of enabledPlaceholders; let index = index">
+    <div [ngClass]="[columnWidths[+ph - 1], columnStyles[+ph - 1] || ''].join(' ').trim()">
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.html:3]: let's compute that inside of "component class" instead, it's too much to keep in markup

File: packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.html [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+      <div [ngClass]="['row']">
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.html:4]: This value is not computed
```suggestion
      <div class="row">
```

File: packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.html [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+        <sc-placeholder
+          [key]="index"
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.html:6]: Not sure if "sc-placeholder" accepts "index", I didn't find this property

File: packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.html [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+          [name]="'column-' + ph + '-{*}'"
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.html:7]: let's compute the placeholder name inside of "component class" instead, you can define a function

File: packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.html [hunk 9]
```diff
@@ -0,0 +1,13 @@
+﻿<div [ngClass]="['row', 'component', 'column-splitter', columnSplitterStyles]" [attr.id]="id || null">
+  <div *ngFor="let ph of enabledPlaceholders; let index = index">
```
Comment (illiakovalenko) [inline packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/column-splitter/column-splitter.component.html:2]: Not sure if "index" is needed, see my comment below

**Review by illiakovalenko — CHANGES_REQUESTED** [review]
See two notes:
1. Please, make sure to verify your changes before opening the PR, since the component is not rendered as expected (I tested locally). I created a bug in our dashboard
image
2. Yes - we are reusing (copying) SXA's components functionality from Next.js. But if we have some room for improvement or ways how to make it more prettier it should be applied

**Review by art-alexeyenko — APPROVED** [review]
Looks good to me. Thanks Addy!

**Pull request merged**